In [11]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import sklearn as svn
%matplotlib inline

from PIL import Image
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import plotly.figure_factory as ff
from prettytable import PrettyTable
from PIL import Image, ImageDraw, ImageFont

In [12]:
data = pd.read_csv('MGH_COVID_Clinical_Info.txt', sep=';')

## Sanity check
**head**
**NA**
**Duplicated**


In [13]:
data.head(2)

subject_id  COVID  Age_cat  BMI_cat  HEART  LUNG  KIDNEY  DIABETES  HTN  \
0           1      1        1        4      0     0       0         0    0   
1           2      1        2        2      0     0       0         0    0   

   IMMUNO  ...  crp_3_cat  ddimer_3_cat  ldh_3_cat  abs_neut_7_cat  \
0       0  ...        1.0           1.0        1.0             NaN   
1       0  ...        2.0           NaN        NaN             NaN   

   abs_lymph_7_cat  abs_mono_7_cat  creat_7_cat  crp_7_cat  ddimer_7_cat  \
0              NaN             NaN          NaN        NaN           NaN   
1              NaN             NaN          NaN        NaN           NaN   

   ldh_7_cat  
0        NaN  
1        NaN  

[2 rows x 44 columns]

In [7]:
data.duplicated().sum()

0

In [14]:
data.isna().sum()

subject_id           0
COVID                0
Age_cat              0
BMI_cat              0
HEART                0
LUNG                 0
KIDNEY               0
DIABETES             0
HTN                  0
IMMUNO               0
Resp_Symp            0
Fever_Sympt          0
GI_Symp              0
D0_draw              1
D3_draw              1
D7_draw              1
DE_draw              1
Acuity_0             0
Acuity_3             0
Acuity_7             0
Acuity_28            0
Acuity_max           0
abs_neut_0_cat       4
abs_lymph_0_cat      5
abs_mono_0_cat       8
creat_0_cat          3
crp_0_cat           18
ddimer_0_cat        26
ldh_0_cat           22
Trop_72h             0
abs_neut_3_cat     106
abs_lymph_3_cat    107
abs_mono_3_cat     110
creat_3_cat        105
crp_3_cat          156
ddimer_3_cat       175
ldh_3_cat          191
abs_neut_7_cat     205
abs_lymph_7_cat    206
abs_mono_7_cat     206
creat_7_cat        193
crp_7_cat          240
ddimer_7_cat       251
ldh_7_cat  

In [15]:
##droping the only one NaN in D0_draw
df = data.drop(383)

### For this study, what is important is to identify the key elements that would affect on patient's prognosis when arrive at the clinic (D0_Draw = 1)

In [16]:
## Generation of dataframe with individuals that were COVID19 positive and started the study at day 0
df=df.loc[(df['COVID']==1)&(df['D0_draw']==1)]

**Dropping the corresponding subject with NaN values.**

In [20]:
df2 = df.dropna(subset=['abs_neut_0_cat','abs_lymph_0_cat',
                        'abs_mono_0_cat','creat_0_cat',
                        'crp_0_cat','ddimer_0_cat',
                       'ldh_0_cat','ldh_0_cat'])

In [21]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 281 entries, 0 to 382
Data columns (total 44 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   subject_id       281 non-null    int64  
 1   COVID            281 non-null    int64  
 2   Age_cat          281 non-null    int64  
 3   BMI_cat          281 non-null    int64  
 4   HEART            281 non-null    int64  
 5   LUNG             281 non-null    int64  
 6   KIDNEY           281 non-null    int64  
 7   DIABETES         281 non-null    int64  
 8   HTN              281 non-null    int64  
 9   IMMUNO           281 non-null    int64  
 10  Resp_Symp        281 non-null    int64  
 11  Fever_Sympt      281 non-null    int64  
 12  GI_Symp          281 non-null    int64  
 13  D0_draw          281 non-null    float64
 14  D3_draw          281 non-null    float64
 15  D7_draw          281 non-null    float64
 16  DE_draw          281 non-null    float64
 17  Acuity_0        

**There are in total 281 individuals that were COVID positive and elegible for this study**

In [22]:
df2.to_csv('COVID_pos.csv')